.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+repos:
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.6.8
+  hooks:
+    - id: ruff
+      args: [ --fix, --exit-non-zero-on-fix ]
+    - id: ruff-format
+      types_or: [ python, pyi]
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: v1.11.2
+  hooks:
+    -   id: mypy
+        args: [--strict, --ignore-missing-imports]
```

Comment:
Please, no pre-commit. It is very strange tool, because it has separated config with different tools versions. It is better to use native git hooks which run command `just lint` for example

Reply:
What about a solution where `pre-commit` invokes `just`:

```yaml
repos:
- repo: local
  hooks:
  - id: lint
    name: lint
    entry: just
    args: [lint]
    language: system
    types: [python] 
    pass_filenames: false
```

My goal is just to avoid those extra linting & formatting commits.

Reply:
seems good